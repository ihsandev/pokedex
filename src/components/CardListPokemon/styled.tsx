import styled from "styled-components";
import { MediaQuery } from "../../utils";

export const MainWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #eaeaea;
  background-color: #fff;
  border-radius: 15px;
  padding: 10px 20px;
  box-sizing: border-box;
  margin-bottom: 10px;
  cursor: pointer;
  transition: all ease 0.2s;
  &:hover {
    box-shadow: 2px 0px 25px #eaeaea;
    transform: scale(1.02);
  }
`;

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  h3 {
    font-weight: bolder;
    text-transform: capitalize;
    margin-left: 30px;
    font-size: 1.2em;
  }
`;

export const ImageView = styled.div`
  text-align: center;
  img {
    max-width: 80px;
    max-height: 80px;
  }
  ${MediaQuery.smMin} {
    img {
      max-width: 100px;
    }
  }
`;

export const Number = styled.div`
  font-size: 1.5em;
  color: #ddd;
  font-weight: bold;
  ${MediaQuery.smMin} {
    font-size: 2em;
  }
`;
