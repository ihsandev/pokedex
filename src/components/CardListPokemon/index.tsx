import React from "react";
import { Icon } from "../../components";
import { Wrapper, ImageView, MainWrapper, Number, Delete } from "./styled";

export interface PropsCard extends React.HTMLAttributes<HTMLDivElement> {
  name?: string;
  src?: string;
  id?: any;
  isDelete?: boolean;
  onDelete?: any;
}
const CardListPokemon: React.FC<PropsCard> = ({
  name,
  src,
  id,
  isDelete,
  onDelete,
  ...props
}) => {
  return (
    <MainWrapper {...props}>
      <Wrapper>
        <ImageView>
          <img src={src} alt="pokemon" />
        </ImageView>
        <div>
          <h3>{name}</h3>
        </div>
      </Wrapper>
      {isDelete ? (
        <Delete onClick={onDelete}>
          <Icon name="ri-delete-bin-2-line" size={30} color="red" />
        </Delete>
      ) : (
        <Number>#{id < 10 ? `0${id}` : id}</Number>
      )}
    </MainWrapper>
  );
};

export default CardListPokemon;
