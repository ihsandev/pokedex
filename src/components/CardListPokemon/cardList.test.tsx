import * as React from "react";
import { shallow } from "enzyme";
import CardListPokemon from "./index";

test("render CardList without props", () => {
  const wrapper = shallow(<CardListPokemon />);
  expect(wrapper).toMatchSnapshot();
});

test("render CardList with props", () => {
  const wrapper = shallow(
    <CardListPokemon
      name="Bulbasaur"
      id={1}
      src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
      onDelete={() => console.log("deleted")}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
