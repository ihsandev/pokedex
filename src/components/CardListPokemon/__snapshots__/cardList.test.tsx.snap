// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CardList with props 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <img
        alt="pokemon"
        src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
      />
    </styled.div>
    <div>
      <h3>
        Bulbasaur
      </h3>
    </div>
  </styled.div>
  <styled.div>
    #
    01
  </styled.div>
</styled.div>
`;

exports[`render CardList without props 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <img
        alt="pokemon"
      />
    </styled.div>
    <div>
      <h3 />
    </div>
  </styled.div>
  <styled.div>
    #
  </styled.div>
</styled.div>
`;
